### DISCUSSION TITLE
Embeds don't seem to be working in Quarto

### DISCUSSION THREAD

**Discussion by ForBo7** [quarto-dev/quarto-cli#2511] [Q&A] [answered]
Hello!

I have a Gradio app hosted on a Huggingspace Space. Gradio offers the ability to embed an app hosted on a Huggingspace Space in another website. The docs for doing so are here: [https://gradio.app/sharing_your_app/#embedding-hosted-spaces](https://gradio.app/sharing_your_app/#embedding-hosted-spaces).

I have a Quarto website (ignore the text on the website in the screenshots below) and wanted to try and embed my app to it. I first do so by using web components.
```html
<script type="module"
src="https://gradio.s3-us-west-2.amazonaws.com/3.3.1/gradio.js">
</script>
<gradio-app space="$ForBo7/$FloodDetector"></gradio-app>
```

I add this snippet to my index.qmd file.

However, when I then preview my site, the Gradio logo along with a loading animation momentarily appears, and then the page goes blank.

<img width="1528" alt="Screen Shot 2022-09-20 at 15 09 01" src="https://user-images.githubusercontent.com/85829792/191231021-341dde8d-d3ec-4281-a690-9f3a1910104e.png">

I then tried using an iframe. I put the following snippet into my index.qmd file.

```html
<iframe src="https://hf.space/embed/$ForBo7/$FloodDetector/+"></iframe>
```

But similarly, nothing appears. However, if I highlight the page, there is an element for the iframe.

<img width="1528" alt="Screen Shot 2022-09-20 at 15 11 48" src="https://user-images.githubusercontent.com/85829792/191231644-538e107c-e366-4335-a979-d4d21e73bf80.png">

I'm using Quarto 1.2.134, macOS Monterey 12.5.1, and an M1 device.

Is there something I'm doing wrong (perhaps I have to add some specific front matter to index.qmd?), is this a bug, or is this something not supported yet?

I appreciate any input!

**Comment by dragonstyle** [comment, marked as answer]
I was able to successfully load their example app both using a script and an embed, using:

```markdown
<iframe src="https://hf.space/embed/abidlabs/pytorch-image-classifier/+"></iframe>

<script type="module"
src="https://gradio.s3-us-west-2.amazonaws.com/3.3.1/gradio.js">
</script>
<gradio-app space="abidlabs/pytorch-image-classifier"></gradio-app>
```

I am guessing that your url includes extraneous `$` characters that should be removed:

```markdown
<script type="module"
src="https://gradio.s3-us-west-2.amazonaws.com/3.3.1/gradio.js">
</script>
<gradio-app space="ForBo7/FloodDetector"></gradio-app>
```

**Reply by ForBo7** [reply]
Ahh, the smallest of mistakes! I got it to work too after removing the `$` character.

Thank you for the response!

**Comment by peter-gy** [comment]
For users looking at embedding Gradio apps into Quarto documents, the previous approach required deploying Gradio applications to a Python server and then embedding the hosted app. With the release of [Gradio Lite](https://www.gradio.app/docs/js/lite), however, Gradio apps can now run completely within the browser.

To simplify the process, I've developed a Quarto extension ([quarto-gradio](https://github.com/peter-gy/quarto-gradio)) that allows you to embed Gradio apps directly into your `qmd` documents or `ipynb` notebooks. Using this extension, you can generate fully serverless Quarto web bundles that include your app as a Gradio Lite bundle.
